diff --git a/llwe.c b/llwe.c
index 9307fc7..01e33be 100644
--- a/llwe.c
+++ b/llwe.c
@@ -121,7 +121,7 @@ loop:	if (isend(end))
 	if (c == 0)
 		r++;
 	end++;
-	if (r < LINES)
+	if (r < LINES - 1)
 		goto loop;
 	else
 		end--;
@@ -147,6 +147,7 @@ static void draw(void)
 
 static void doscrl(int d)
 {
+	clear();
 	lwe_scroll += d;
 	if (lwe_scroll < 0)
 		lwe_scroll = 0;
@@ -255,7 +256,7 @@ static int insertmode(char *t)
 			c = '\n';
 		if (c == C_D)
 			return 1;
-		if (c == KEY_BACKSPACE) {
+		if (c == '\b' || c == 127) {
 			if (t <= buffer)
 				continue;
 			t--;
@@ -293,14 +294,10 @@ static int cmdloop(void)
 		c = getch();
 		switch (c) {
 		case C_D:
-		case KEY_DOWN:
-		case KEY_NPAGE:
 		case 'j':
 			doscrl(LINES / 2);
 			break;
 		case C_U:
-		case KEY_UP:
-		case KEY_PPAGE:
 		case 'k':
 			doscrl(-LINES / 2);
 			break;
@@ -369,8 +366,6 @@ int main(int argc, char **argv)
 		cbreak();
 		noecho();
 		nonl();
-		intrflush(stdscr, FALSE);
-		keypad(stdscr, TRUE);
 		filename = argv[1];
 		ed();
 		endwin();
