diff --git a/src/llwe.c b/src/llwe.c
index 505bb77..973fd5b 100644
--- a/src/llwe.c
+++ b/src/llwe.c
@@ -65,7 +65,7 @@ static bool initbuf(int sz)
 
 static bool filetobuf(int sz)
 {
-	FILE *f = fopen(filename, "r");
+	FILE *f = fopen(filename, "r", "ctx=stm");
 	if (f == NULL) {
 		err("read");
 		return false;
@@ -104,7 +104,7 @@ int breload()
 
 static int bsave(void)
 {
-	FILE *f = fopen(filename, "w");
+	FILE *f = fopen(filename, "w", "ctx=stm");
 	if (f == NULL) {
 		const char msg1[] = "Error: Failed to write to: ";
 		const char msg2[] = "Press any key to continue.";
@@ -146,7 +146,7 @@ static void winbounds(void)
 	if (c == 0)
 		r++;
 	end++;
-	if (r < LINES)
+	if (r < LINES - 1)
 		goto loop;
 	else
 		end--;
@@ -172,6 +172,7 @@ static void draw(void)
 
 static void doscrl(int d)
 {
+	clear();
 	lwe_scroll += d;
 	if (lwe_scroll < 0)
 		lwe_scroll = 0;
@@ -399,7 +400,7 @@ static int insertmode(char *t)
 			c = '\n';
 		if (c == C_D)
 			return 1;
-		if (c == KEY_BACKSPACE) {
+		if (c == '\b' || c == 127) {
 			if (t <= buffer)
 				continue;
 			t--;
@@ -601,12 +602,8 @@ static enum loopsig lineoverlaycmd(void)
 
 static command_fn cmdtbl[512] = {
 	[C_D] = scrolldown,
-	[KEY_DOWN] = scrolldown,
-	[KEY_NPAGE] = scrolldown,
 	['j'] = scrolldown,
 	[C_U] = scrollup,
-	[KEY_UP] = scrollup,
-	[KEY_PPAGE] = scrollup,
 	['k'] = scrollup,
 	['q'] = quitcmd,
 	['i'] = insertcmd,
@@ -656,8 +653,6 @@ int main(int argc, char **argv)
 		cbreak();
 		noecho();
 		nonl();
-		intrflush(stdscr, FALSE);
-		keypad(stdscr, TRUE);
 		filename = argv[1];
 		ed();
 		endwin();
