diff --git a/src/llwe.c b/src/llwe.c
index e6b2987..9e30e53 100644
--- a/src/llwe.c
+++ b/src/llwe.c
@@ -71,7 +71,7 @@ static bool initbuf(int sz)
 
 static bool filetobuf(int sz)
 {
-	FILE *f = fopen(filename, "r");
+	FILE *f = fopen(filename, "r", "ctx=stm");
 	if (f == NULL) {
 		err("read");
 		return false;
@@ -110,7 +110,7 @@ int breload()
 
 static int bsave(void)
 {
-	FILE *f = fopen(filename, "w");
+	FILE *f = fopen(filename, "w", "ctx=stm");
 	if (f == NULL) {
 		const char msg1[] = "Error: Failed to write to: ";
 		const char msg2[] = "Press any key to continue.";
@@ -164,7 +164,7 @@ static void winbounds(void)
 {
 	int r = 0, c = 0;
 	start = skipscreenlines(buffer, lwe_scroll);
-	for (end = start; !isend(end) && r < LINES; end++) {
+	for (end = start; !isend(end) && r < LINES - 1; end++) {
 		c++;
 		if (*end == '\n')
 			c = 0;
@@ -196,6 +196,7 @@ static void draw(void)
 
 static void doscrl(int d)
 {
+	clear();
 	lwe_scroll += d;
 	if (lwe_scroll < 0)
 		lwe_scroll = 0;
@@ -342,7 +343,7 @@ static void drawlinelbls(int lvl, int off)
 	int count = 0;
 	char *p = start;
 	int toskip = off;
-	for (int line = 0; line < LINES;) {
+	for (int line = 0; line < LINES - 1;) {
 		if (toskip == 0) {
 			move(line, 0);
 			ptarg(count++);
@@ -435,7 +436,7 @@ static int insertmode(char *t)
 			c = '\n';
 		if (c == C_D)
 			return 1;
-		if (c == KEY_BACKSPACE) {
+		if (c == '\b' || c == 127) {
 			if (t <= buffer)
 				continue;
 			t--;
@@ -633,7 +634,7 @@ static enum loopsig lineoverlaycmd(void)
 	int screenline = 0;
 	int fileline = 0;
 	attron(A_STANDOUT);
-	while (screenline < LINES) {
+	while (screenline < LINES - 1) {
 		char nstr[32];
 		snprintf(nstr, sizeof(nstr), "%4d", lineno);
 		mvaddstr(screenline, 0, nstr);
@@ -652,12 +653,8 @@ static enum loopsig lineoverlaycmd(void)
 
 static command_fn cmdtbl[512] = {
 	[C_D] = scrolldown,
-	[KEY_DOWN] = scrolldown,
-	[KEY_NPAGE] = scrolldown,
 	['j'] = scrolldown,
 	[C_U] = scrollup,
-	[KEY_UP] = scrollup,
-	[KEY_PPAGE] = scrollup,
 	['k'] = scrollup,
 	['q'] = quitcmd,
 	['i'] = insertcmd,
@@ -707,8 +704,6 @@ int main(int argc, char **argv)
 		cbreak();
 		noecho();
 		nonl();
-		intrflush(stdscr, FALSE);
-		keypad(stdscr, TRUE);
 		filename = argv[1];
 		ed();
 		endwin();
