diff --git a/src/llwe.c b/src/llwe.c
index b9d5196..fb3e1f2 100644
--- a/src/llwe.c
+++ b/src/llwe.c
@@ -70,7 +70,7 @@ bread(void)
 		err("memory");
 		return 0;
 	}
-	f = fopen(filename, "r");
+	f = fopen(filename, "r", "ctx=stm");
 	if (f == NULL) {
 		return 1;
 	}
@@ -93,7 +93,7 @@ breload()
 static int
 bsave(void)
 {
-	FILE *f = fopen(filename, "w");
+	FILE *f = fopen(filename, "w", "ctx=stm");
 	if (f == NULL) {
 		err("write");
 		return 0;
@@ -129,7 +129,7 @@ loop:	if (isend(end))
 	if (c == 0)
 		r++;
 	end++;
-	if (r < LINES)
+	if (r < LINES - 1)
 		goto loop;
 	else
 		end--;
@@ -158,6 +158,7 @@ draw(void)
 static void
 doscrl(int d)
 {
+	clear();
 	lwe_scroll += d;
 	if (lwe_scroll < 0)
 		lwe_scroll = 0;
@@ -300,7 +301,7 @@ insertmode(char *t)
 			c = '\n';
 		if (c == C_D)
 			return 1;
-		if (c == KEY_BACKSPACE) {
+		if (c == '\b' || c == 127) {
 			if (t <= buffer)
 				continue;
 			t--;
@@ -441,12 +442,8 @@ jumptolinecmd(void)
 
 static command_fn cmdtbl[512] = {
 	[C_D] = scrolldown,
-	[KEY_DOWN] = scrolldown,
-	[KEY_NPAGE] = scrolldown,
 	['j'] = scrolldown,
 	[C_U] = scrollup,
-	[KEY_UP] = scrollup,
-	[KEY_PPAGE] = scrollup,
 	['k'] = scrollup,
 	['q'] = quitcmd,
 	['i'] = insertcmd,
@@ -493,8 +490,6 @@ main(int argc, char **argv)
 		cbreak();
 		noecho();
 		nonl();
-		intrflush(stdscr, FALSE);
-		keypad(stdscr, TRUE);
 		filename = argv[1];
 		ed();
 		endwin();
