diff --git a/llwe.c b/llwe.c
index 0e6aded..3a24f28 100644
--- a/llwe.c
+++ b/llwe.c
@@ -66,7 +66,7 @@ static int bread(void)
 		err("memory");
 		return 0;
 	}
-	f = fopen(filename, "r");
+	f = fopen(filename, "r", "ctx=stm");
 	if (f == NULL) {
 		return 1;
 	}
@@ -87,7 +87,7 @@ int breload()
 
 static int bsave(void)
 {
-	FILE *f = fopen(filename, "w");
+	FILE *f = fopen(filename, "w", "ctx=stm");
 	if (f == NULL) {
 		err("write");
 		return 0;
@@ -121,7 +121,7 @@ loop:	if (isend(end))
 	if (c == 0)
 		r++;
 	end++;
-	if (r < LINES)
+	if (r < LINES - 1)
 		goto loop;
 	else
 		end--;
@@ -147,6 +147,7 @@ static void draw(void)
 
 static void doscrl(int d)
 {
+	clear();
 	lwe_scroll += d;
 	if (lwe_scroll < 0)
 		lwe_scroll = 0;
@@ -279,7 +280,7 @@ static int insertmode(char *t)
 			c = '\n';
 		if (c == C_D)
 			return 1;
-		if (c == KEY_BACKSPACE) {
+		if (c == '\b' || c == 127) {
 			if (t <= buffer)
 				continue;
 			t--;
@@ -318,14 +319,10 @@ static int cmdloop(void)
 		c = getch();
 		switch (c) {
 		case C_D:
-		case KEY_DOWN:
-		case KEY_NPAGE:
 		case 'j':
 			doscrl(LINES / 2);
 			break;
 		case C_U:
-		case KEY_UP:
-		case KEY_PPAGE:
 		case 'k':
 			doscrl(-LINES / 2);
 			break;
@@ -397,8 +394,6 @@ int main(int argc, char **argv)
 		cbreak();
 		noecho();
 		nonl();
-		intrflush(stdscr, FALSE);
-		keypad(stdscr, TRUE);
 		filename = argv[1];
 		ed();
 		endwin();
